Add some replays stuff into build

Zert0X opened the pull request:
#1577
ХОЧУ ТЕСТОВЫЙ МЕРЖ
![изображение](https://user-images.githubusercontent.com/42303535/123547370-29ef0400-d769-11eb-9fae-4dea9a7d7443.png)


Видосик:
https://streamable.com/680ds6

Известные на данный момент проблемы:
1. Спрайты хуманов.
2. Дверцы шкафов 
3. Оверлей лампочек
4. При фолове моба вьюевер начинает лагать, предположительно из-за того что ?

- [ ] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Zert0X committed:
Demo init

Zert0X committed:
Demo init

Zert0X committed:
demo-viewer

Zert0X committed:
demo-viewer

Zert0X committed:
Turf and objects demo logging

Zert0X committed:
Turf and objects demo logging

Zert0X committed:
admins confidentiality

Zert0X committed:
admins confidentiality

Zert0X committed:
Some more confidentiality

Zert0X committed:
Some more confidentiality

Zert0X committed:
Change some configs

Zert0X committed:
Change some configs

Zert0X commented:
примерно 43 тысячи это приблуда для просмотра этих реплеев

TobyThorne commented:
Весёлый способ помножить лаги на два

Doster-d commented:
Ты ебанутый блять

Zert0X commented:
> Весёлый способ помножить лаги на два

Ну в оригинале говорили, что не повышает

Zert0X commented:
> Ты ебанутый блять

А я что? Сказали портировать, я и портировал

igorsaux commented:
вхере original?

Doster-d commented:
Откуда порт?

igorsaux commented:
yogstation понял

TobyThorne commented:
я это мержить без плина и эпикуса точно не буду
на первый взгляд - тысячи оверхеда и усложнение кода ради довольно сомнительной фичи

igorsaux commented:
ревертнуть всегда успеем

Zert0X committed:
Oops

Zert0X committed:
Oops

Zert0X committed:
demo-viewer into tools

Zert0X committed:
demo-viewer into tools

Commit:
tweak(Exodus): fixes assorted stuff, adds new shuttle chairs subtype

PR #5739 
- Удалена документация психоскопа с Исхода.
- Обновлен эвакуационный шаттл "Исхода" - удален стол из-под консоли, заменены некоторые клетки пола.
- Добавлен органфиксер воксам и нюкерам.
- Обновлен шаттл шахтеров - стулья заменены на новые кресла.
- Обновлен шаттл синдиката - стулья заменены на кресла, исправлена планировка кабины пилота.

TatarTheBuilder committed:
tweak(Frontier): adds new lightbulb boxes, fixes some bugs

PR #5735
- На Фронтире теперь лежат коробки с новыми лампами.
- Изменена мусорка в ксенобио на Фронтире.

No-Tips committed:
tweak(killer tomatoes): tweaks movespeed

PR #5708

RdEErAqUa committed:
tweak(tajaran): tajaran overheat while wearing spacesuits

PR #5421

TatarTheBuilder committed:
tweak(light fixtures): replaces hanging lantern with old light fixtures

PR #5715

Doster-d committed:
tweak(DBBan): shows the banned player a notification

PR #5766

TobyThorne committed:
tweak(flashbangs): reworks flashbangs

PR #5719 
- Нерфобафф светошумовых гранат. Меньше стана и нокдауна, больше ослепления и оглушения.

TobyThorne committed:
tweak(uplink): changes nuclear ops' TC amount to 15, some prices go up, reworks assault rifles

PR #5769 
- Переработана штурмовая винтовка 5.56мм. Нарисованы новые спрайты. Название изменено с STS-35 на AS-75. Вместимость магазина увеличена с 20 до 25 патронов. Урон пулек 5.56мм увеличен с 25 до 27.5.
- Запас кристаллов в аплинке нюкеров повышен с 12 до 15 ТК.
- Повышены цены на некоторое оружие нюкеров чтобы скомпенсировать увеличенный запас.
- В аплинк нюкеров добавлен пулемёт, L6 SAW (за 12 ТК) и коробка на 60 патронов для него (за 3 ТК).
- Подкорректирован флешдарк. Теперь темнота от него накладывается сильнее и не отдаёт синим.
- Исправлена невозможность покупки хамелеон-бомбы в аплинке.
- Очевиднее названы некоторые позиции патронов в аплинке (добавлены приписки "Magazine", "Speedloader", etc).

NefrNerf committed:
fix(adminpaper): fixes user access to admin fax

PR #5669

igorsaux committed:
Changelog 26/06/21 (#5772)

Zert0X committed:
Mob sprites on replays

FUCK YEAH!!111!!!

Zert0X committed:
Mob sprites on replays

FUCK YEAH!!111!!!

Zert0X committed:
Some More Confidentiality

Zert0X committed:
Some More Confidentiality

Zert0X commented:
Итак предлагаю тестовый мёрдж ради познания насколько будет лагать сервак и сколько будет весить один реплей!

MaxFreemen commented:
> Итак предлагаю тестовый мёрдж ради познания насколько будет лагать сервак и сколько будет весить один реплей!

Аминь.

Zert0X committed:
idk

Zert0X committed:
idk

Zert0X committed:
Doster optimization

Zert0X committed:
Doster optimization

RdEErAqUa committed:
tweak(tajaran): tajaran overheat while wearing spacesuits, but now without damage

PR #5782

igorsaux committed:
fix(build): fixes 3 runtimes

PR #5801

fixes #5800
fixes #5799
fixes #5797

igorsaux committed:
tweak(vending machines): changes NanoUI to TGUI

PR #5713 
- Редизайн интерфейса торговых автоматов.

Shirlineyn committed:
fix(build): fixes assorted borg-related bugs, allows drag-n-dropping ppl into resleevers, allows traitorborgs to emag themselves

PR #4820 
- Cервис-грипперы теперь могут в кухонное оборудование.
- Грипперы теперь могут вставлять батарейки в PSUшки.
- Дропперы теперь нельзя перемолоть в гриндере.
- В ресливер теперь можно класть мобов драг'н'дропом.
- Дроны теперь могут в надувные стены.
- Киборги-триторы могут емагнуть себя сами, используя соответствующий верб.

TobyThorne committed:
fix(icons): centers a number of icons

PR #5808

igorsaux committed:
fix(build): adds icon2base64 for better icon-HTML handling

PR #5811

TapKo4eB committed:
fix(water_ballon): w_size approximation & fulfillment from dispensers

PR #5709 
- Водяные шарики теперь приобрели ожидаемый малый размер и научились менять его при наполнении.
- Водяные шарики теперь можно наполнять не только из водяных баков.

Zert0X committed:
Some shity mob thing revert

Zert0X committed:
Some shity mob thing revert

TobyThorne committed:
fix(medicine): fixes carotid artery typo

PR #5806

igorsaux committed:
tweak(chat): adds graphic emoji support

PR #5819

No-Tips committed:
fix(corgi): allows corgies to wear hats... again

PR #5698 
- Корги снова могут носить шапки.

TobyThorne committed:
tweak(events): reduces viruses' weight, makes some events non-oneshot

PR #5818 
- Уменьшен вес всех трёх уровней вирусов: Space Cold Outbreak (20 -> 15 за медика), Viral Infection (40 -> 20 за медика), Plague Infection (35 -> 20 за медика). Plague Infection теперь одноразовый ивент.
- Space Vines, Carp Migration, Carp School, Appendicitis, Lost Carp - больше не одноразовые ивенты.
- Вес Appendicitis снижен (15 -> 3 за медика), ибо теперь может произойти больше одного раза за раунд.
- Вес Wormholes увеличен (10 по умолчанию -> 10 по умолчанию + 1 за каждого игрока).

NefrNerf committed:
fix(wizard): teleporting from cryo no longer produces cringe

PR #5750 
- Добавлена проверка на нахождении в криокапсуле, если маг кастует из нее, то его сначала оттуда выкинет, и после этого произойдет блинк/телепорт.

TobyThorne committed:
fix(build): allows the build to compile

PR #5820 
- Исправлена ошибка с инклюдами из одного из ПРов.

igorsaux committed:
Changelog 29/06/21 (#5821)

Artemonim committed:
fix(viruses): Runtime in helpers.dm, 92: Cannot read null.name

PR #5822

Epicus7 committed:
feat(docs): add "as anything()" section to contributing.dm

TobyThorne committed:
tweak(backpacks): renews medical, sci and hydro backpacks worn icons

PR #5840 
-  Обновлены спрайты некоторых рюкзаков (у медиков, химиков, вирусологов и учёных) на спине. Теперь они выглядят соответственно спрайтам самих рюкзаков.

igorsaux committed:
fix(vending.dm): Runtime in vending.dm, 188: Cannot execute null.GetIdCard().

PR #5836

NefrNerf committed:
fix(mimics): defaults their obj type to crate 

PR #5841 
- Заспавненные мимики теперь по умолчанию будут ящиками, так что больше никаких бессмертных ебак.

Artemonim committed:
tweak(consoles): engineering consoles have nanoword preinstalled

PR #5827

NefrNerf committed:
fix(wizard): fixes knock and contract skills

PR #5848 
- Заклинание "Knock" открывает обесточенные шлюзы.
- Исправлены заклинания взаимодействия мага с учениками.
- Заклинание "Return to Master" телепортирует ученика к магу, а не наоборот.

TobyThorne committed:
fix(stun gloves): fixes broken initialize

PR #5865 
- Стан-перчатки инициализировались через жопу.
- Заспавненные/намаппленные стан-перчатки теперь по умолчанию выбирают базой рабочие перчатки. До этого - получалось поломанное нечто.
- Попытка обрезать/обмотать проводами перчатки-хамелеоны теперь выдаёт очевидное сообщение, как у боксёрских перчаток.

igorsaux committed:
tweak(R&D machinery): changes plain browser to TGUI

PR #5859

TobyThorne committed:
tweak(sfx): cuts a delay before the actual sound in smash.ogg

PR #5883

TobyThorne committed:
fix(build): fixes skulls dupe, some weird xenobehavior, reentering qdeleted bodies, etc

PR #5891 
- Исправлена возможность бесконечно спавнить черепа и ошмётки с помощью голов.
- Из голов дион и ксеноморфов больше нельзя доставать человеческие черепа.
- Добавлена задержка при ударах по мехам.
- Грудоломы теперь появляются через секунду после разрывания человека. Чтобы их не убивало летящими запчастями тел в тесных помещениях.
- При использовании Re-Enter Corpse добавлена проверка на нахождение тела в нуллспейсе. Поскольку сейчас практически никакие мобы нормально не куделятся, попытка вернуться в такого моба приводила ко всяким паранормальным явлениям вроде появления в виде мозга/хэдкраба-зомби в космосе/на астероиде/в жопе админа.
- Исправлена возможность появления у королевы ксеноморфов нейролейса.
- Исправлено превращение дрона в грудолома вместо королевы при использовании Evolve, если дрон появился не "натуральным" путём, а с помощью кривого использования админских кнопок.
- Чутка ослабил петлю на шее у GC, прибравшись в Destroy() у мобов. Большая часть мобов всё равно ещё где-то как-то референсится, но по крайней мере /mob/living/simple_animal уже нормально удаляются, что радует.
- Теперь при получении роли ксеноморфа будут выводиться разные сообщения в зависимости от того, является ли моб грудоломом или ксеноморфом.

TobyThorne committed:
fix(ammunition): fixes some missing ammo icons

PR #5881

TobyThorne committed:
tweak(vending machines): makes UI buttons thinner

PR #5867

igorsaux committed:
tweak(fonts): removes Shage

PR #5871 
...whatever it was

Epicus7 committed:
tweak(storyteller): adds more roles to storyteller

PR #5500 
- Добавляем больше ролей в сторителлера. Поправил рантайм при переключении вкладок панели управления сторителлером.
- При спавне борера, мага и ниндзи сторителлер опрашивает гостов и игроков в лобби с соответствующим префенсом, хотят ли они быть ролью. Если игрок отклоняет или не отвечает в течение 10 секунд, то опрашивается следующий игрок.
- Нужно еще допилить борерам перенос тушки КУДА-ТО - пока не придумал где их спавнить можно.

TobyThorne committed:
fix(floor): fixes floor tiles dupe

PR #5900 
- Исправлена возможность дюпать плитку, отдирая её с тайла, под которым есть стена.

Zert0X commented:
Я кажись придумал как починить мобов, приеду, завтра, проверу

TobyThorne committed:
fix(tape recorder): fixes broken icon_states

PR #5906 
- Диктофоны больше не невидимые.

Doster-d committed:
fix(build): items can no longer be thrown out of vents, banned players see some info

PR #5903 
- Из труб больше нельзя бросать предметы.
- Теперь при заходе с баном, помимо причины, выдается ссылка на то, куда писать, чтобы подать апелляцию.

Zert0X committed:
Some shity mob thing revert x2

Zert0X committed:
Some shity mob thing revert x2

Zert0X committed:
Merge branch 'YEET_Replays' of https://github.com/Zert0X/OnyxBay into YEET_Replays

Zert0X commented:
Ухожу в драфт

TermService commented:
![изображение](https://user-images.githubusercontent.com/18397924/124398495-a3b15f80-dd1e-11eb-9475-301b6e7fc85f.png)

Doster-d commented:
А закрывать-то зачем...

Zert0X commented:
> 
> 
> А закрывать-то зачем...

так надо сына, так надо. А вообще я просто наебнул ветку дважды
